extension/observer/observer.go

```diff
@@ -36,16 +36,22 @@ const (
 
 // Observable is an interface that provides notification of endpoint changes.
 type Observable interface {
-	// TODO: Stopping.
 	// ListAndWatch provides initial state sync as well as change notification.
-	// notify.OnAdd will be called one or more times if there are endpoints discovered.
+	// notify. OnAdd will be called one or more times if there are endpoints discovered.
 	// (It would not be called if there are no endpoints present.) The endpoint synchronization
 	// happens asynchronously to this call.
 	ListAndWatch(notify Notify)
+
+	// Unsubscribe stops the previously registered Notify from receiving callback invocations.
+	Unsubscribe(notify Notify)
```

Comment:
Should there be an error returned if you try Unsubscribed a Notify that isn't present?

Reply:
I followed the existing callback centric design and none were provided by `ListAndWatch()`. It does seem like a black box and error handling could make sense in `Notify` though.

Reply:
Happy to leave that as an improvement for a future PR, doesn't sound like it is needed atm.